frontend/src/components/dashboard/campaigns/announcement-modal/AnnouncementModal.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useContext, useState, useEffect } from 'react'
+import cx from 'classnames'
+import { updateAnnouncementVersion } from 'services/settings.service'
+
+import { PrimaryButton, ErrorBlock } from 'components/common'
+import { ModalContext } from 'contexts/modal.context'
+
+import styles from './AnnouncementModal.module.scss'
+import { i18n } from 'locales'
+import { ANNOUNCEMENT } from 'config'
+import { OutboundLink } from 'react-ga'
+
+const AnnouncementModal = () => {
+  const { close, setBeforeClose } = useContext(ModalContext)
+  const [errorMessage, setErrorMessage] = useState('')
+
+  useEffect(() => {
+    setBeforeClose(() => () => {
+      updateAnnouncementVersion(i18n._(ANNOUNCEMENT.version))
```

Comment:
You will need to handle the error thrown if `updateAnnouncementVersion` fails.

Reply:
done!